### ISSUE TITLE
Requesting for Addressbook access

### ISSUE THREAD

**Issue by rahuljiresal** [Alterplay/APAddressBook#44]
Currently, if the app does not have access to the Address Book, it never asks for it. It only returns `APAddressBookAccessUnknown` on `[APAddressBook access]`.

You should add an API for requesting the access to the address book. Something like

`+ (void)requestAccessWithCompletionBlock:(void(^)(void)block;`

**Comment by mrpatiwi** [comment]
+1

**Comment by eriknelson** [comment]
This should probably at least be mentioned in the documentation. +1

**Comment by belkevich** [comment]
Check out version 0.1.10

**Comment by bgd011** [comment]
Apple's comments for "ABAddressBookRequestAccessWithCompletion" say
"...
// The user will only be prompted the first time access is requested; any subsequent calls
// to ABAddressBookCreateWithOptions will use the existing permissions.
..."
So I don't think any subsequent calls to it will do anything. I have tested the new requestAccess method and it does not seem to do anything.

**Comment by belkevich** [comment]
It runs block with existing permissions after first call. 
Sorry, looks like I don't understand your question. What exactly do you mean?

**Comment by bgd011** [comment]
I just meant that I tested requestAccess method vs just loadContacts and I am not sure of the benefits of first checking for access and calling requestAccess if it is not granted. 
Both methods call ABAddressBookRequestAccessWithCompletion and this method will behave the same. 
So calling loadContacts pretty much does the same as requestAccess plus more. Maybe some documentation that explains when to use the requestAccess method, because I did not find a use case.
